great_expectations/core/expectation_configuration.py

```diff
@@ -1402,6 +1403,22 @@ def metrics_validate(
             execution_engine=execution_engine,
         )
 
+    def get_domain_type(self) -> Optional[MetricDomainTypes]:
+        """Return "domain_type" of this expectation."""
+        if self.expectation_type.startswith("expect_table_"):
+            return MetricDomainTypes.TABLE
+
+        if "column" in self.kwargs:
+            return MetricDomainTypes.COLUMN
+
+        if "column_A" in self.kwargs and "column_B" in self.kwargs:
```

Comment:
Should this be case insensitive?

Reply:
good catch @cdkini. Same question from me

Reply:
@cdkini these are exactly how our `kwargs` keys were defined, so would not work otherwise.  Thanks.

Reply:
Not sure how maintainable this is though. Do our kwargs follow some kind of schema? Outside of reading this PR, how can I know that "column_A" is a valid key?

Reply:
I think we could use `typing.TypeDict`  for this.

https://docs.python.org/3.10/library/typing.html#typing.TypedDict

The limitations on the `typing_extensions` version shouldn't be an issue.
https://github.com/python/typing_extensions#other-notes-and-limitations

Reply:
@cdkini For us, those keys have always been in the code, since V1, then in V2, and in V3.  It could be by convention.  You can see them in the API documentation, too (at least in the previous version).  Thanks.